src/sentry/models/apitoken.py

```diff
@@ -252,6 +267,64 @@ def update(self, *args: Any, **kwargs: Any) -> int:
     def outbox_region_names(self) -> Collection[str]:
         return list(find_all_region_names())
 
+    def _should_flush_outbox(self) -> bool:
+        from sentry import options
+
+        has_async_flush = self.user_id in options.get("users:api-token-async-flush")
+        logger.info(
+            "async_flush_check",
+            extra={
+                "has_async_flush": has_async_flush,
+                "user_id": self.user_id,
+                "token_id": self.id,
+            },
+        )
+        if has_async_flush:
+            return False
+
+        return True
+
+    @contextlib.contextmanager
+    def _maybe_prepare_outboxes(self, *, outbox_before_super: bool) -> Generator[None]:
+        # Overriding to get around how default_flush cannot be cleanly feature flagged
+        flush = self._should_flush_outbox()
```

Comment:
in order to FF the default_flush property change, we need to override `_maybe_prepare_outboxes` since default_flush cannot be dynamically set